### PR TITLE
Timeout documentation.

### PR DESCRIPTION
**Pull request by dmitrievanthony**

### CODE REVIEW

File: clients/http-client/features/timeout.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  artifact: io.ktor:ktor-client-core:$ktor_version
+  class: io.ktor.client.features.HttpTimeout
+ktor_version_review: 1.4.0
+---
+
+Timeout allows limiting the time of the request execution or execution steps such as a connection or a TCP packet awaiting. The following timeout types are available:
+
+* __request timeout__ — time-bound of the whole request processing,
+* __connect timeout__ — time-bound of the connection establishment,
+* __socket timeout__ — time-bound of the interval between packet reception.
```
Comment (cy6erGn0m) [inline clients/http-client/features/timeout.md]: Not exactly clear. "Between X and Y" or "it is far between." But here it looks like something is missing in the sentence

File: clients/http-client/features/timeout.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Be default all these timeouts are infinite and should be explicitly specified when needed. Timeouts could be specified for all requests of a particular client or for a single request.
```
Comment (cy6erGn0m) [inline clients/http-client/features/timeout.md]: By

File: clients/http-client/features/timeout.md (hunk 1 above)
Comment (dmitrievanthony) [reply clients/http-client/features/timeout.md]: Updated, please have a look.

File: clients/http-client/features/timeout.md (hunk 2 above)
Comment (dmitrievanthony) [reply clients/http-client/features/timeout.md]: Thanks, fixed.

**Pull request merged**